chapter04/GamblersProblem.py

```diff
@@ -20,9 +20,13 @@
 # optimal policy
 policy = np.zeros(GOAL + 1)
 
-# state value
+# reward Value
+rewardValue = np.zeros(GOAL + 1)
+rewardValue[GOAL] = 1.0
+
+# state Value
 stateValue = np.zeros(GOAL + 1)
-stateValue[GOAL] = 1.0
```

Comment:
All rewards are 0 except transition to the GOAL state (a dummy state). So we can set the value of goal state to 1 and make sure we don't update the goal state during the iterations. If we do so, all the rewards become 0 so we can just ignore the rewards. See Exercise 4.9